src/check/name/true_name/mod.rs

```diff
@@ -19,13 +18,17 @@ pub mod python;
 pub struct TrueName {
     pub is_nullable: bool,
     pub is_mutable: bool,
-    pub variant: NameVariant,
+    pub variant: StringName,
 }
 
 pub trait IsTemp {
     fn is_temp(&self) -> bool;
 }
 
+pub trait MatchTempName {
+    fn temp_map(&self, other: &StringName, mapping: HashMap<Name, Name>, pos: Position) -> TypeResult<HashMap<Name, Name>>;
```

Comment:
Hm perhaps some of this logic could be re-used for `match_name`?
They are very similar, except for the if statement.
So if you can pass a predicate then they're the same basically.

Reply:
Do this at the end perhaps when we have a build where all the tests pass again.
Will make it much easier to quickly verify the refactor.